src/Microsoft.ML.Data/Utilities/ModelFileUtils.cs

```diff
@@ -283,8 +283,8 @@ internal static void SaveRoleMappings(IHostEnvironment env, IChannel ch, RoleMap
             {
                 // REVIEW: Should really validate the schema here, and consider
                 // ignoring this stream if it isn't as expected.
-                var loader = TextLoader.ReadFile(env, new TextLoader.Arguments(),
-                    new RepositoryStreamWrapper(rep, DirTrainingInfo, RoleMappingFile));
+                var loader = new TextLoader(env, dataSample: new RepositoryStreamWrapper(rep, DirTrainingInfo, RoleMappingFile))
+                    .Read(new RepositoryStreamWrapper(rep, DirTrainingInfo, RoleMappingFile));
```

Comment:
We shouldn't create two instances of `new RepositoryStreamWrapper(rep, DirTrainingInfo, RoleMappingFile)`.

Reply:
Maybe it may be valuable to still have an internal `TextLoader.ReadFile` helper method for our internal code.

---
In reply to: [239856914](https://github.com/dotnet/machinelearning/pull/1784#discussion_r239856914) [](ancestors = 239856914)